src/dune/exe.ml

```diff
@@ -36,6 +36,10 @@ module Linkage = struct
 
   let js = { mode = Byte; ext = ".bc.js"; flags = [] }
 
+  let is_plugin = function
+    | { ext = ".cma" | ".cmxs"; _ } -> true
```

Comment:
I would need to extend `Link_mode.t`, roughly as follows:
```
type t =
  | Byte
  | Native
  | Byte_with_stubs_statically_linked_in
  (* This two are new *)
  | Plugin_native
  | Plugin_byte
```
it it worth it? By the way we also match on the extension in the case of JS compilation.

Reply:
Seems like it's worth it. Although I would do `Plugin of Mode.t` instead.

But actually, you could also just do:

```ocaml
let is_plugin t = List.mem t.ext [Mode.plugin_ext Byte; Mode.plugin_ext Native]
```

Not very beautiful, but it's best to avoid having all these extensions everywhere.

The issue with the .js extension is indeed something that we need to fix.

Reply:
I took the second suggestion, it was a less invasive change.